api/preprints/parsers.py

```diff
@@ -3,11 +3,12 @@
 
 class PreprintsJSONAPIParser(JSONAPIParser):
     def flatten_relationships(self, relationships):
-        ret = super(PreprintsJSONAPIParser, self).flatten_relationships(relationships)
-        related_resource = relationships.keys()[0]
-        if ret.get('target_type') and ret.get('id'):
-            return {related_resource: ret['id']}
-        return ret
+        rel = {}
+        for resource in relationships:
+            ret = super(PreprintsJSONAPIParser, self).flatten_relationships({resource: relationships[resource]})
+            if ret.get('target_type') and ret.get('id'):
```

Comment:
As I recall, you're very limited with the information you have here, but this would be so much better if it got the relationship name rather, or in addition to, the target_type. This way you could have two relationships with the same type.

Also, this would probably close another ticket if it were the base flatten_relationships, but we can just make a note on that ticket to do what we did here if we need to.